src/smt.cpp

```diff
@@ -14,7 +14,32 @@ z3::expr_vector toExprVector(const vector<smt::expr> &vec) {
 }
 
 namespace smt {
+class Context {
+
+public:
+  std::optional<z3::context> z3_ctx;
```

Comment:
No std:: in .cpp

Reply:
Should we remove *all* `std::` in the .cpp file? (Like move, reverse, nullopt, ...)